### PR TITLE
bug-bash: Change phase to directive in custom logic doc

### PR DESCRIPTION
**Pull request by Guaris**
Fixes https://github.com/Kong/docs.konghq.com/issues/7533

### CODE REVIEW

File: app/_src/gateway/plugin-development/custom-logic.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ of {{site.base_gateway}}'s execution life-cycle:
 - **[HTTP Module]** *is used for plugins written for HTTP/HTTPS requests*
 {% if_version lte: 3.3.x %}
 
-| Function name       | Phase             | Request Protocol        | Description
+| Function name       | Directive             | Request Protocol        | Description
```
Comment (lena-larionova) [inline app/_src/gateway/plugin-development/custom-logic.md:45]: I think it's a bit more complicated than this. The items in this column cover both a phase and a directive. Check out the diagram here: https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#directives.



As the reporter points out in the GH issue,
* "rewrite" in Kong -> stands for the "rewrite tail" phase and the "rewrite_by_lua_*" directive: https://github.com/openresty/lua-nginx-module#rewrite_by_lua_block

It looks like what we're actually missing is another column, which details what the directive is. So for rewrite, you get:

Function: rewrite
Phase: rewrite tail
Directives: `rewrite_by_lua_*`

**Review by lena-larionova — CHANGES_REQUESTED** [review]
This is missing the second part of the issue as well, which is:

> An other issue concerns the "reponse" function, for which the phase is "access". However, according to the description, it replaces "header_filter()" and "body_filter()". As shown in the diagram in the [Directives](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#directives) section, they are part of "Content Phase", not "Access Phase". This seems to be a mistake.

Edit: After doing more digging, I found that the `response` function runs in the Kong `response` phase, as a whole separate thing: https://github.com/Kong/kong/blob/master/spec/fixtures/custom_plugins/kong/plugins/enable-buffering-response/handler.lua. The links to the directives are correct though (header_filter and body_filter). 

See all the Kong phases here: https://github.com/Kong/kong/blob/master/kong/pdk/private/phases.lua

File: app/_src/gateway/plugin-development/custom-logic.md (hunk 1 above)
Comment (lena-larionova) [reply app/_src/gateway/plugin-development/custom-logic.md:45]: Did a bunch more digging, and this isn't quite right either. It seems that Kong has come up with its own terminology here, where "access phase" or "init_worker phase" are accurate _only for Kong_. So - technically this column _is_ correctly named "phase", but the links to directives make it confusing.

**Review by lena-larionova — APPROVED** [review]
tl;dr for the changes I made to this PR:
* marked ws phases as enterprise only
* added a directives column and renamed the original phases column to "Kong phases", which are not the same thing as the nginx phases
* checked each function against its real phase in Kong; the only one that actually differs from its function name is the `configuration` function
